### ISSUE TITLE
Prelude and Serenade cannot be foolish when using "Randomize Overworld Spawns"

### ISSUE THREAD

**Issue by chimpanreeve** [TestRunnerSRL/OoT-Randomizer#1360]
As per Roman, you need access to places with refills when using this setting, and right now that can only be Kakariko Village and Kokiri Forest.

As such, when not using another form of ER, these songs will never been required with just this setting alone. Fixing this would mean that locations with songs can be foolish again.

**Comment by ETR-BTF** [comment]
Poldon suggested it might be nice to "implement a check to see if each warp song destination is reachable shere 0 from both spawn points, and if it is then it's foolish." Though I don't think we've had items before that could be foolish or not foolish depending on the seed (rather than the settings).

**Comment by Zannick** [comment]
No, items are only foolish if they are always foolish based on the settings alone.

With no other ER settings, I expect access to Kakariko to grant unrestricted access to Hyrule Field and thus to Castle Town/Temple of Time and Lake Hylia. So these items should be foolish under `spawn_positions` but that's listed among the ER settings that make them not foolish.

https://github.com/TestRunnerSRL/OoT-Randomizer/blob/95500b80357cd1b1ca9e7da47019be58f75e9e24/World.py#L666-L667
Seems like a simple change to fix.

**Comment by r0bd0g** [comment]
I think the reason they're not able to be foolish currently is because in theory they shouldn't be able to be, it's just that weird quirk with requiring no item access (for item refills) to kokiri or kakariko that prevents it. And so we're pretending the handling of refills is more robust than it really is. Maybe what should be changed is the ability to spawn somewhere where those songs actually could be useful.
